AirCasting/SDSync/Parsing/CSVSession.swift

```diff
@@ -0,0 +1,55 @@
+// Created by Lunar on 09/12/2021.
+//
+
+import Foundation
+import CoreLocation
+
+class CSVSession {
+    
+    private(set) var sessions = Set<SDSession>()
+    private(set) var streamsWithMeasurements: [SDStream: [Measurement]] = [:]
+    
+    private let parser = SDCardMeasurementsParser()
+    
+    init(fileURL: URL, fileLineReader: FileLineReader) throws {
```

Comment:
so this class is basically getting a file and parsing it into our structures `SDSession`, `SDStream`, `SDMeasurement`. So maybe it should be called `SDFileParser` and also for DI purposes be split into protocol and implementation. If we do so we can then inject `FileLineReader` into implementation using init like we do with other dependencies there and instead of doing all the paring in `init` we just create a func for this:
```Swift
protocol SDFileParser {
    func parse(file: URL) throws -> (sessions: [SDSession], streams: [SDStream: [Measurement]])
}

class DefaultSDFileParser: SDFileParser {
    ...
    init(fileReader: FileLineReader) { }
    ...
}
```